### ISSUE TITLE
Cert behavior regression (2.14.0)

### ISSUE THREAD

**Issue by romuald** [psf/requests#5563]
## TLDR;

requests used to ignore the cert parameter when used, now it raises an error when the certificate files are missing

-------

I'm certainly late to the party, but we found a regression introduced in requests 2.14.0

From the changelog:
> When users provide invalid paths to certificate bundle files or directories Requests now raises IOError, rather than failing at the time of the HTTPS request with a fairly inscrutable certificate validation error

When used with **http://** URLs, the previous behavior was to ignore the certificate completely.

"Now" requests will always raise an error regardless of the protocol

## Reproduction Steps

```python
import requests

req = requests.get('http://perdu.com', cert=('nope.crt', 'nope.key'))
print(req)
```

Works fine on 2.13.0, and raise an IOError with requests 2.14.0 and up (since the certificates does not exist)

As fixing this might introduce more regressions, feel free to close it as invalid

**Comment by nateprewitt** [comment]
Hi @romuald, thanks for reporting this. We made this change intentionally due to the unintuitive failure mode prior to 2.14.0. Given it's been in place for a few year and this is the first feedback we've received about it, I think we'll leave it in place.
